src/v/http/demo/run.sh

```diff
@@ -30,6 +30,20 @@ if [[ -f "./build/debug/clang/bin/http_demo_client" ]]; then
   demo_client_bin=${demo_client_bin:-"./build/debug/clang/bin/http_demo_client"}
 fi
 
+# ./vbuild is a build path used by vtools
+if [[ -f "./vbuild/release/gcc/bin/http_demo_client" ]]; then
+  demo_client_bin="./vbuild/release/gcc/bin/http_demo_client"
+fi
+if [[ -f "./vbuild/release/clang/bin/http_demo_client" ]]; then
+  demo_client_bin=${demo_client_bin:-"./vbuild/release/clang/bin/http_demo_client"}
+fi
+if [[ -f "./vbuild/debug/gcc/bin/http_demo_client" ]]; then
+  demo_client_bin=${demo_client_bin:-"./vbuild/debug/gcc/bin/http_demo_client"}
+fi
+if [[ -f "./vbuild/debug/clang/bin/http_demo_client" ]]; then
+  demo_client_bin=${demo_client_bin:-"./vbuild/debug/clang/bin/http_demo_client"}
```

Comment:
this doesn't make sense to me. 

don't we need to short circuit. otherwise it says

1. use gcc
2. use clang
3. use gcc debug
4. use clang debug

so if you have all 4 builds locally, it will always pick clang debug. 

is that what we want? seem confusing to me. also, couldn't this be solved by a for loop 

for i in { options } ;  if  x; break; done.

Reply:
if this test is wrapped in task, it can make use of `BUILD_DIR` to point to the right place to find the binaries.